kani-compiler/src/codegen_aeneas_llbc/mir_to_ullbc/mod.rs

```diff
@@ -0,0 +1,799 @@
+        }
+
+        trace!("{:?}", name);
+        Ok(Name { name })
+    }
+
+    /// Compute the span information for the given instance
+    fn translate_instance_span(&mut self, instance: Instance) -> CharonSpan {
+        self.translate_span(instance.def.span())
+    }
+
+    /// Compute the span information for MIR span
+    fn translate_span(&mut self, span: Span) -> CharonSpan {
+        let filename = FileName::Local(PathBuf::from(span.get_filename()));
+        let file_id = match self.translated.file_to_id.get(&filename) {
+            Some(file_id) => *file_id,
+            None => {
+                let file_id = self.translated.id_to_file.push(filename.clone());
+                self.translated.file_to_id.insert(filename, file_id);
+                file_id
+            }
+        };
+        let lineinfo = span.get_lines();
+        let rspan = RawSpan {
+            file_id,
+            beg: Loc { line: lineinfo.start_line, col: lineinfo.start_col },
+            end: Loc { line: lineinfo.end_line, col: lineinfo.end_col },
+            rust_span_data: rustc_internal::internal(self.tcx(), span).data(),
+        };
+
+        // TODO: populate `generated_from_span` info
+        CharonSpan { span: rspan, generated_from_span: None }
+    }
+
+    fn translate_function_signature(&mut self) -> FunSig {
+        let instance = self.instance;
+        let fn_abi = instance.fn_abi().unwrap();
+        let requires_caller_location = self.requires_caller_location(instance);
+        let num_args = fn_abi.args.len();
+        let args = fn_abi
+            .args
+            .iter()
+            .enumerate()
+            .filter_map(|(idx, arg_abi)| {
+                // We ignore zero-sized parameters.
+                // See https://github.com/model-checking/kani/issues/274 for more details.
+                // We also ingore the last parameter if the function requires
+                // caller location.
+                if arg_abi.mode == PassMode::Ignore
+                    || (requires_caller_location && idx + 1 == num_args)
+                {
+                    None
+                } else {
+                    let ty = arg_abi.ty;
+                    debug!(?idx, ?arg_abi, "fn_typ");
+                    Some(self.translate_ty(ty))
+                }
+            })
+            .collect();
+
+        debug!(?args, ?fn_abi, "function_type");
+        let ret_type = self.translate_ty(fn_abi.ret.ty);
+
+        // TODO: populate the rest of the information (`is_unsafe`, `is_closure`, etc.)
+        FunSig {
+            is_unsafe: false,
+            is_closure: false,
+            closure_info: None,
+            generics: GenericParams::default(),
+            parent_params_info: None,
+            inputs: args,
+            output: ret_type,
+        }
+    }
+
+    fn translate_function_body(&mut self) -> Result<BodyId, Opaque> {
+        let instance = self.instance;
+        let mir_body = instance.body().unwrap();
```

Comment:
From our offline chat. I think what you want is the following, you only want to generate code for generic code, but you want to use the reachability code to collect everything that is reachable from the user input. In that case, I recommend you do the following.

1. After the collector runs, build a set of `CrateDef` from all the collected instances. You can convert an instance to a `CrateItem` using the `TryFrom` implementation [here](https://github.com/rust-lang/rust/blob/baaf3e65abab5a281715230bcb1785610629ae7d/compiler/stable_mir/src/mir/mono.rs#L214-L226).
2. Most places where you use instance, you can replace by the crate item. I think the only place that you will need to change is where you the the `fn_abi()`. In that case, you need to use the function signature instead.